liferay-echo/liferay-portal
# [TS] LPS-180284 The validation message of some inputs doesn't mention the correct input name

Motivation
=======
This issue ([LPS-180284](https://issues.liferay.com/browse/LPS-180284)) was reported by a customer for a particular example but the issue can be reproduced in different places in the portal. In detail, the validation for a field shown with `<aui:input>` doesn't show the correct message: it mentions the `name` instead of the `label` field.

There are problematic uses of the tag in several modules:
- account
- blogs
- captcha
- commerce
- fragment
- iframe
- layout
- login
- portal-instances
- portal-language
- portal-settings
- portal-web
- portal-workflow
- server
- site-navigation
- sytle-book

I've sent this first to @liferay-echo because some of its modules are affected and the changes applied are in the style of [LPS-167115](https://issues.liferay.com/browse/LPS-167115). 

Please let me know if I should pass the PR around to the different teams or if a review by one team is enough since the changes are the same everywhere.

Proposed Solution
========

To use the `required` field directly in the taglib `<aui:input>` instead of an inner `validator` with the special name `required`. This way the message displayed when the field is not filled in uses the `label` instead of the `name`.

This is strictly necessary when the `name` and the `label` attributes in `<aui:input>` are different. (First commit c0ad1cbc511f0638c9bfe5a4233820550ffd4467.)

When `name` and `label` aren't the same there's no visual problem the second commit (87ee9919c11cbee6c8c1c9de63cd26a44e31dbf2) applies a similar change to avoid problems in the future when `name` and `label` might differ.

A similar case, that of `label` not being explicitly set, is addressed in the third commit (ff5b2653a428493fdf2273da46651ea6ae897747).


Steps to Verify
========
Those in [LPS-180284](https://issues.liferay.com/browse/LPS-180284) for the particular example of the Create Account form, but similar ones exist for all files address in the first commit (c0ad1cbc511f0638c9bfe5a4233820550ffd4467).

For example, one involving Echo would be:
1. Create a new Navigation Menu.
2. Add an URL item.
3. Without filling out any of the fields (Name and URL), click on the Add button.
Expected: The validation message for URL reads "The URL field is required.".
Observed: The validation message for URL reads "The type-settings-properties--url-- field is required.".


References to existing code
========

The same solution was applied for [LPS-167115](https://issues.liferay.com/browse/LPS-167115) [here](https://github.com/liferay-echo/liferay-portal/blob/master/modules/apps/asset/asset-categories-admin-web/src/main/resources/META-INF/resources/category/details.jsp#L72-L74).

Checklist (optional)
========

CODE:
- [x] I have considered breaking this PR into smaller PRs if the amount of changed code is too large (Although many files are changed, the changes are the same throughout so I decided to create just one PR.)

COMMITS:
- [x] There are no commits which should be combined into others (Maybe just one commit would have sufficed but this way might be cleaner.)

PR TITLE & DESCRIPTION:
